src/core/common/settings.hpp

```diff
@@ -584,7 +584,9 @@ class Settings : public SettingsBase
      */
     otError ReadSlaacIidSecretKey(Utils::Slaac::IidSecretKey &aKey)
     {
-        return ReadFixedSize(kKeySlaacIidSecretKey, &aKey, sizeof(Utils::Slaac::IidSecretKey));
+        uint16_t length = sizeof(Utils::Slaac::IidSecretKey);
+
+        return Read(kKeySlaacIidSecretKey, &aKey, length);
```

Comment:
Should it return error if the read length is smaller?

Reply:
The length check has been removed from all the other reads, which implies that we trust the settings driver implementation to have some value and length that we expect. Is there a use case here that does not apply to the other settings?

Reply:
It can be helpful when debugging, but I'm fine with this assumption.